openmetadata-ui/src/main/resources/ui/src/hooks/paging/usePaging.ts

```diff
@@ -17,12 +17,21 @@ import {
   useMemo,
   useState,
 } from 'react';
+import { useHistory } from 'react-router-dom';
 import {
   INITIAL_PAGING_VALUE,
   PAGE_SIZE_BASE,
   pagingObject,
 } from '../../constants/constants';
+import { CursorType } from '../../enums/pagination.enum';
 import { Paging } from '../../generated/type/paging';
+import useCustomLocation from '../useCustomLocation/useCustomLocation';
+
+interface CursorState {
+  cursorType: CursorType | null;
+  cursorValue: string | null;
+  currentPage: number | null;
```

Comment:
currentPage is already there so no need to pull it here sepratly

Reply:
currentPage present in the hook will get reset to initial value when the page is reloaded. That's why i have store currentPage's value in the history state to retrive it whenever the user has paginated and clicked back or the page is reloaded.